src/core/ai/llm/messages/manager.ts

```diff
@@ -525,7 +525,7 @@ export class ContextManager {
         try {
             // Use pre-computed system prompt if provided
             const prompt = systemPrompt ?? (await this.getSystemPrompt(context));
-            return this.formatter.format([...messageHistory], prompt);
+            return this.formatter.format([...messageHistory], prompt, context);
```

Comment:
trusting that this works cuz we don't have tests, let's just validate this manually